internal/am/delete_transfer.go

```diff
@@ -0,0 +1,45 @@
+package am
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+
+	"github.com/artefactual-sdps/enduro/internal/sftp"
+)
+
+const DeleteTransferActivityName = "DeleteTransferActivity"
+
+type DeleteTransferActivityParams struct {
+	Destination string
+}
+
+type DeleteTransferActivity struct {
+	client sftp.Client
+	logger logr.Logger
+}
+
+func NewDeleteTransferActivity(logger logr.Logger, client sftp.Client) *DeleteTransferActivity {
+	return &DeleteTransferActivity{client: client, logger: logger}
+}
+
+func (a *DeleteTransferActivity) Execute(ctx context.Context, params *DeleteTransferActivityParams) error {
+	a.logger.V(1).Info("Execute DeleteTransferActivity",
+		"SourcePath", params.Destination,
+	)
+
+	dest, err := os.Open(params.Destination)
+	if err != nil {
+		return fmt.Errorf("Delete transfer: %v", err)
+	}
+	defer dest.Close()
+
+	err = a.client.Delete(ctx, params.Destination)
+	if err != nil {
+		return fmt.Errorf("Delete transfer: %v", err)
```

Comment:
I think it would be good to include the remote file path in the error message too, so an operator can manually find and delete the transfer if necessary.